### ISSUE TITLE
Request::body_json should return 4xx error if body can not be deserialized

### ISSUE THREAD

**Issue by jhorstmann** [http-rs/tide#494]
Currently the `body_json` function returns a `std::io::Result`, which gets converted into a 500 status code later. While this is probably ok for actual io errors, any errors happening during parsing of the json bytes are more likely caused by incorrect data being send by the client and so should result in a 400 Bad Request status code.

See https://github.com/http-rs/tide/blob/999c30c65b26c8d4b9210a95523bdcdc8843f8e5/src/request.rs#L266

**Comment by vkill** [comment]
you can use [status](https://docs.rs/tide/0.8.1/tide/trait.Status.html#tymethod.status) custom it in your app

```rust
use tide::Status;

let body = serde_json::from_slice(&body_bytes).status(400)?
```

**Comment by jannik4** [comment]
This should be fixed since #537, so I think this can be closed.
